lmdeploy/turbomind/turbomind.py

```diff
@@ -53,47 +56,81 @@ def _tm_dict_to_torch_dict(tm_dict: _tm.TensorMap):
     return ret
 
 
+@contextmanager
+def cuda_ctx(device_id):
+    old_device = torch.cuda.current_device()
+    torch.cuda.set_device(device_id)
+    yield
+    torch.cuda.set_device(old_device)
+
+
 class TurboMind:
     """LMDeploy's inference engine.
 
     Args:
         model_path (str): the path of turbomind's model
         data_type (str): the data type
-        session_len (int): the max length of a session
         eos_id (int): eos token id
         stop_words (List[int]): token ids of stop-words
-        device_id (int): the id of a gpu card
-        node_id (int): the id of a node
-        device_num (int): the number of gpu cards
-        node_num (int): the number of node
     """
 
     def __init__(self,
                  model_path: str,
                  data_type: str = 'fp16',
-                 session_len: int = 2048,
                  eos_id: int = 2,
-                 stop_words: List[int] = None,
-                 device_id: int = 0,
-                 node_id: int = 0,
-                 device_num: int = 1,
-                 node_num: int = 1):
+                 stop_words: List[int] = None):
         self.eos_id = eos_id
 
-        # create model instance
+        # mpi
+        mpi_comm = MPI.COMM_WORLD
```

Comment:
turbomind only supports TP on one node,  so we can set node_id 0 and node_num 1. In this way, turbomind python API can avoid 'mpi4py' dependency

Reply:
nccl ids is broadcasted with mpi.
https://github.com/InternLM/lmdeploy/blob/1f88baa5b7a9dde22b11200fd530fe1059e1facb/src/turbomind/triton_backend/transformer_triton_backend.cpp#L43
 I have to bind `ft::mpi::initialize` if we remove mpi4py

Reply:
Done

Reply:
We can remove nccl id broadcasting part in `transformer_triton_backend.cpp`. They are useless in TP